### PR TITLE
Fix SBOM behavior on forks

### LINKED ISSUE #4680: Azure Pipeline Validation failed with "The signing feature is not available for your organization yet." at step "Generate BOM"
**Describe the bug**
Pipeline Validation failed with "The signing feature is not available for your organization yet." at step "Generate BOM".
I have retried for several times and get the same failure.

***Exception or Stack Trace***
You can get the detail failure in this PR: https://github.com/Azure/azure-sdk-for-cpp/pull/4664
And the failed pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2815063&view=logs&jobId=7c6ba154-1dd3-507d-6d24-8db90b0372c9&j=7c6ba154-1dd3-507d-6d24-8db90b0372c9&t=01245f84-53cd-51f5-7f46-ca8f9eb2dd03

image

### PR DESCRIPTION
**Pull request by danieljurek**
Fixes #4680

Example pipeline runs: 

* PR run internal (no SBOM created) -- https://dev.azure.com/azure-sdk/public/_build/results?buildId=2819369&view=logs&j=e96721ea-85b6-589d-07ca-a23ddc86e9a2&t=2f94eae2-9bab-5a20-cdeb-6a7bbc189f60
* PR run public (no SBOM created) -- https://dev.azure.com/azure-sdk/public/_build/results?buildId=2819365&view=logs&j=e96721ea-85b6-589d-07ca-a23ddc86e9a2&t=2f94eae2-9bab-5a20-cdeb-6a7bbc189f60
* Manual run internal (SBOM is created) -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2819370&view=logs&j=e96721ea-85b6-589d-07ca-a23ddc86e9a2&t=7b42e275-1077-5f51-c1c2-656cd96d055e

### CODE REVIEW

File: eng/pipelines/templates/steps/sbom.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+parameters:
```
Comment (benbp) [inline eng/pipelines/templates/steps/sbom.yml:1]: I think this could be renamed to `generate-and-publish-sbom.yml`

**Review by weshaggard — APPROVED** [review]
Change looks good. We may want to consider moving this to eng/common and have the other language repos use it as well. For now though I'm good with it here.

**Pull request merged**